pkg/storaged/lvol-tabs.jsx

```diff
@@ -508,3 +508,77 @@ export class PoolVolTab extends React.Component {
         );
     }
 }
+
+export const VDOPoolTab = ({ client, lvol }) => {
+    const vdo_iface = client.vdo_vols[lvol.path];
+    const vdo_pool_vol = client.lvols[vdo_iface.VDOPool];
+
+    function grow() {
+        lvol_grow(client, vdo_pool_vol, { });
+    }
+
+    function toggle_compression() {
+        return new Promise((resolve, reject) => {
+            const new_state = !vdo_iface.Compression;
+            vdo_iface.EnableCompression(new_state, {})
+                    .catch(reject);
+            client.wait_for(() => vdo_iface.Compression === new_state)
```

Comment:
What about chaining the waiting with the method call?

```
return vdo_iface.EnableCompression(new_state, {})
   .then(() => client.wait_for(() => vdo_iface.Compression === new_state));
```

client.wait_for returns a promise (that delivers the return value of the condition callback), so we can just chain these up.

Reply:
I thought of that initially, but then was worried about race conditions -- that the property *might* already be correct after EnableCompression() returns, and then there would not be a direct property update signal afterwards any more.

However, that does look more elegant for sure, so I did that now. I ran it locally a few times (where I could hardly reproduce the original error that CI caught), and it seems fine. Let's see what the bots think.

Reply:
Seems they are all happy :green_heart:

Reply:
> I thought of that initially, but then was worried about race conditions -- that the property _might_ already be correct after EnableCompression() returns, and then there would not be a direct property update signal afterwards any more.

`wait_for` should handle that.  It checks the condition once immediately, and then on every change notification.